# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmInitExpr.cpp
# Commit: 662cb63c42d7
# Full Hash: 662cb63c42d7874a79256a6edf1a03d83e43a014
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2023-08-05 09:19:01
# Regressor Bug: 1692065
# File Overlap Count: 2
# Description:
#   Bug 1692065 - wasm: Create a unique wasm::AnyRef type. r=yury,jonco
#   
#   Right now wasm references area represented as a JSObject*. This is assumed in
#   many places in SpiderMonkey, most prominently in our GC integration
#   around tracing, the store buffer, and write barriers. This will cause issues
# ==============================================================================

diff -r 8b506ab41451 -r 662cb63c42d7 js/src/wasm/WasmInitExpr.cpp
--- a/js/src/wasm/WasmInitExpr.cpp	Fri Aug 04 13:54:57 2023 +0000
+++ b/js/src/wasm/WasmInitExpr.cpp	Fri Aug 04 14:06:49 2023 +0000
@@ -361,7 +361,7 @@
     }
 
     return pushRef(RefType::fromTypeDef(&typeDef, false),
-                   AnyRef::fromJSObject(structObj));
+                   AnyRef::fromJSObject(*structObj));
   }
 
   bool evalStructNewDefault(JSContext* cx, uint32_t typeIndex) {
@@ -373,7 +373,7 @@
 
     const TypeDef& typeDef = instance().metadata().types->type(typeIndex);
     return pushRef(RefType::fromTypeDef(&typeDef, false),
-                   AnyRef::fromJSObject(structObj));
+                   AnyRef::fromJSObject(*structObj));
   }
 
   bool evalArrayNew(JSContext* cx, uint32_t typeIndex) {
@@ -390,7 +390,7 @@
 
     const TypeDef& typeDef = instance().metadata().types->type(typeIndex);
     return pushRef(RefType::fromTypeDef(&typeDef, false),
-                   AnyRef::fromJSObject(arrayObj));
+                   AnyRef::fromJSObject(*arrayObj));
   }
 
   bool evalArrayNewDefault(JSContext* cx, uint32_t typeIndex) {
@@ -403,7 +403,7 @@
 
     const TypeDef& typeDef = instance().metadata().types->type(typeIndex);
     return pushRef(RefType::fromTypeDef(&typeDef, false),
-                   AnyRef::fromJSObject(arrayObj));
+                   AnyRef::fromJSObject(*arrayObj));
   }
 
   bool evalArrayNewFixed(JSContext* cx, uint32_t typeIndex,
@@ -424,7 +424,7 @@
 
     const TypeDef& typeDef = instance().metadata().types->type(typeIndex);
     return pushRef(RefType::fromTypeDef(&typeDef, false),
-                   AnyRef::fromJSObject(arrayObj));
+                   AnyRef::fromJSObject(*arrayObj));
   }
 #endif  // ENABLE_WASM_GC
 };